src/jsdocUtils.js

```diff
@@ -186,20 +186,26 @@ const potentiallyEmptyNamepathTags = [
   'event',
   'callback',
   'class', 'constructor',
+  'extends', 'augments',
   'constant', 'const',
   'function', 'func', 'method',
   'interface',
   'member', 'var',
   'mixin',
   'namespace',
-  'listens', 'fires', 'emits'
+  'listens', 'fires', 'emits',
+  'see',
+
+  // GCC syntax allows typedef to be named through variable declaration rather than jsdoc name
+  'typedef'
```

Comment:
Let's add a global `settings.jsdoc.mode` with possible values of "gcc"|"jsdoc"|"typescript" and use that to determine whether to allow here (if later versions of jsdoc change this, we can add values, e.g., "jsdoc4").

Reply:
This is what I was hoping for in https://github.com/gajus/eslint-plugin-jsdoc/issues/306, but I think it should be done separately from this PR as it aims to fix a bug and introducing a big feature would clutter it and complicate accepting it.

Reply:
Ok, sure, makes sense, though let's make a mental note of that being one item relevant for targeting.